### PR TITLE
Remove core checkout

### PR DESCRIPTION
**Pull request by mbj**
Depends on #626

### CODE REVIEW

File: core/spec/models/spree/order_spec.rb [hunk 1]
```diff
@@ -980,25 +476,49 @@ def apply
...
-        expect { apply }.to change(order, :state).from('delivery').to('cart')
+        expect { apply }
+          .to change(order, :state)
+          .from('delivery').to('cart')
       end
 
       it 'removes shipping method' do
-        expect { apply }.to change { order.shipping_method } .from(shipping_method).to(nil)
+        expect { apply }
+          .to change { order.shipping_method }
```
Comment (dkubb) [inline core/spec/models/spree/order_spec.rb]: This can be written as `.to change(order, :shipping_method)`.

File: core/app/models/spree/order.rb [hunk 2]
```diff
@@ -336,16 +267,14 @@ def empty!
...
+      self.attributes = {
+        state:           'cart',
+        shipping_method: nil,
+        shipment_total:  ZERO_COST
+      }
 
-    def shipping_eq_billing_address?
-      (bill_address.empty? && ship_address.empty?) || bill_address.same_as?(ship_address)
+      update!
+      self
```
Comment (dkubb) [inline core/app/models/spree/order.rb]: As mentioned elsewhere I think `Spree::Order#update!` should return `self` and not the meaningless return value it does now.

Comment (mbj) [inline core/app/models/spree/order.rb]: I think this is the correct PR to expand this fix with.

File: core/spec/models/spree/order_spec.rb (hunk 1 above)
Comment (dkubb) [inline core/spec/models/spree/order_spec.rb]: Bump. This flag is probably hidden in the collapsed-by-default diff of this file.

**Comment by mbj** [comment]
Closing this as we have a superior solution in the main app.

**Pull request closed without merging**